packages/did/src/DidResolver/DidResolver.ts

```diff
@@ -168,47 +132,46 @@ export async function resolveKey(
 export async function resolveServiceEndpoint(
   serviceUri: DidResourceUri
 ): Promise<ResolvedDidServiceEndpoint | null> {
-  const { fragment: serviceId, type, did } = parseDidUri(serviceUri)
+  const { fragment: serviceId, did, type } = parseDidUri(serviceUri)
 
   // A fragment (serviceId) IS expected to resolve a service endpoint.
   if (!serviceId) {
     throw new SDKErrors.InvalidDidFormatError(serviceUri)
   }
 
-  switch (type) {
-    case 'full': {
-      const serviceEndpoint = await queryServiceEndpoint(serviceUri, serviceId)
-      if (!serviceEndpoint) {
-        return null
-      }
-      return {
-        id: serviceUri,
-        type: serviceEndpoint.type,
-        serviceEndpoint: serviceEndpoint.serviceEndpoint,
-      }
+  if (type === 'full') {
+    const endpoint = await queryServiceEndpoint(serviceUri, serviceId)
+    if (!endpoint) {
+      return null
     }
-    case 'light': {
-      const resolvedDetails = await resolve(did)
-      if (!resolvedDetails) {
-        throw new SDKErrors.InvalidDidFormatError(serviceUri)
-      }
-      if (!resolvedDetails.details) {
-        return null
-      }
-      const serviceEndpoint = Did.getEndpoint(
-        resolvedDetails.details,
-        serviceId
-      )
-      if (!serviceEndpoint) {
-        return null
-      }
-      return {
-        id: serviceUri,
-        type: serviceEndpoint.type,
-        serviceEndpoint: serviceEndpoint.serviceEndpoint,
-      }
+    return {
+      ...endpoint,
+      id: serviceUri,
     }
-    default:
-      throw new SDKErrors.UnsupportedDidError(did)
+  }
+
+  const resolved = await resolve(serviceUri)
+  if (!resolved) {
+    return null
+  }
+
+  const {
+    details,
+    metadata: { canonicalId },
+  } = resolved
+  const controller = canonicalId || did
+  const latestDid = canonicalId ? await Did.query(canonicalId) : details
```

Comment:
again, that seems to me just unnecessary. The upgraded full did even having the same service endpoints with the same ids and so on, is rather a niche case, which we would not have to support I think.

Reply:
I have updated the PR to match the outcomes of the discussion